### PR TITLE
fix(#59): align CAM, IAM, CPD, gateway SSO and S2S auth with tm1py

### LINKED ISSUE #59: RestService — Fix CAM, IAM, CPD, and gateway SSO authentication flows
## Phase

Phase 1 (v2.0.0) — Foundation

### Problem

tm1npm's authentication flows diverge from tm1py. CAM auth currently POSTs JSON instead of building `CAMPassport`/`CAMNamespace` Authorization headers. IBM Cloud IAM token exchange, CPD access token generation, and gateway SSO support for CAM are all missing entirely.

### Required Changes

- Fix CAM auth to build `CAMPassport`/`CAMNamespace` Authorization headers instead of POSTing JSON
- Implement IBM Cloud IAM token exchange (POST to `iam_url` with API key, receive bearer token)
- Add CPD access token generation flow
- Add gateway SSO support for CAM authentication
- Ensure all auth flows match tm1py's header construction and token lifecycle

### Files

- `src/services/RestService.ts`

### Reference

- tm1py `RestService.py` lines 1165–1214 (CAM, IAM, CPD auth flows)
- tm1py `RestService.py` lines 1381–1390 (gateway SSO support)

### PR DESCRIPTION
**Pull request by hsubebe89**
## Summary
- **Fix CAM auth**: Replace JSON POST with `Authorization: CAMPassport` / `CAMNamespace` headers (tm1py parity)
- **Add IBM Cloud IAM token exchange**: POST URL-encoded form to `iamUrl`, receive bearer token (`_generateIbmIamCloudAccessToken`)
- **Add CPD access token generation**: POST JSON credentials to `cpdUrl/v1/preauth/signin` (`_generateCpdAccessToken`)
- **Add gateway SSO**: GET gateway URL with `CAMNamespace` param, extract `cam_passport` cookie (`_buildAuthorizationTokenCamSso`)
- **Add PA Proxy auth flow**: Generate CPD JWT → POST to auth URL → extract `ba-sso-csrf` cookie (`_authenticateWithPaProxy`)
- **Fix Service-to-Service**: Use Basic auth with `clientId:clientSecret`, POST `{User: user}` to auth endpoint
- **Rewrite `getAuthenticationMode()`**: Topology-based dispatch matching tm1py's `_determine_auth_mode`
- **Add `decodeB64` support**: Base64-encoded passwords decoded before token construction
- **Propagate `verify` setting**: All external auth requests honor TLS verification config via `insecureAgentOption()`
- **Extract shared helpers**: `normaliseSetCookie()`, `extractCookieValue()`, `insecureAgentOption()` eliminate duplication
- **40+ new test cases**: All auth flows, mode detection, edge cases, error paths

Closes #59

## Test plan
- [x] `npx tsc --noEmit` — no type errors
- [x] `npx jest src/tests/restService.test.ts` — 119 tests passing (40+ new)
- [x] `npx jest` — full suite passes (4 credential-dependent integration suites expected to skip)
- [ ] Manual verification against a TM1 instance with CAM authentication
- [ ] Manual verification against IBM Cloud with IAM token exchange
- [ ] Manual verification against CPD/PA Proxy environment

### CODE REVIEW

File: src/services/RestService.ts [hunk 1]
```diff
@@ -513,192 +513,348 @@ export class RestService {
...
+            case 'base_url':
+            default: {
+                // v11 / base_url: check auth-specific config flags
+                if (c.accessToken) return AuthenticationMode.ACCESS_TOKEN;
+                if (c.apiKey) return AuthenticationMode.BASIC_API_KEY;
+                if (c.applicationClientId && c.applicationClientSecret) {
+                    return AuthenticationMode.SERVICE_TO_SERVICE;
+                }
+                if (c.camPassport) return AuthenticationMode.CAM;
+                if (c.gateway) return AuthenticationMode.CAM_SSO;
```
Comment (hsubebe89) [inline src/services/RestService.ts]: **P1: Missing `namespace` guard for CAM_SSO detection**

This line triggers `CAM_SSO` when only `gateway` is set, but `namespace` is not checked. Downstream in `setupAuthentication()`, `this.config.namespace!` is used with a non-null assertion, which would pass `undefined` to the gateway URL if namespace is not configured.

Either add `&& c.namespace` here:
```typescript
if (c.gateway && c.namespace) return AuthenticationMode.CAM_SSO;
```

Or add a validation guard in the `CAM_SSO` case of `setupAuthentication()`.

Also add a test: config with `gateway` but no `namespace` should throw a clear error (or fall through to a different auth mode).

**Review by hsubebe89 — COMMENTED** [review]
## Summary

Thorough refactor of RestService authentication to align with tm1py's auth flows. The PR replaces the old flat if-else chain with topology-based dispatch, adds proper CAM/CAMNamespace/gateway SSO/IBM IAM/CPD/PA Proxy/S2S flows, and includes 478 new test lines (119 tests passing). The architecture is significantly improved. One missing validation guard needs to be addressed before merging.

## Status: :warning: NEEDS CHANGES

## Scores
- Security: 8/10
- Performance: 9/10
- Code Quality: 9/10
- Test Coverage: 8/10

## Issues Found

### P1 - Important (:warning: NEEDS CHANGES)

1. **[src/services/RestService.ts:761+818] CAM_SSO detection does not require `namespace`, but `setupAuthentication` asserts it non-null**
   - Issue: `getAuthenticationMode()` at line 761 triggers `CAM_SSO` when only `gateway` is set (`if (c.gateway) return AuthenticationMode.CAM_SSO`), without checking that `namespace` is also present. Then in `setupAuthentication()` at line 818, `this.config.namespace!` is passed to `_buildAuthorizationTokenCamSso()` with a non-null assertion. If a user provides `gateway` without `namespace`, this sends `CAMNamespace=undefined` as a query parameter to the gateway URL, producing a confusing error rather than a clear validation message.
   - Impact: Misconfigured clients get an opaque gateway failure instead of an actionable error message. The non-null assertion (`!`) silences TypeScript's type safety.
   - Fix: Either (a) add `&& c.namespace` to the `CAM_SSO` detection in `getAuthenticationMode()`, or (b) add a guard at the top of the `CAM_SSO` case in `setupAuthentication`:
     ```typescript
     case AuthenticationMode.CAM_SSO: {
         if (!this.config.gateway || !this.config.namespace) {
             throw new Error('CAM SSO authentication requires both gateway and namespace');
         }
         const token = await RestService._buildAuthorizationTokenCamSso(
             this.config.gateway,
             this.config.namespace,
             this.config.verify
         );
         // ...
     }
     ```
     Add a test: `gateway` set without `namespace` should throw a clear error.

### P2 - Suggestions (:bulb: Optional)

1. **[src/services/RestService.ts:800-805] PA_PROXY non-null assertions on `user` and `password`** - `password!` could be `undefined` if the user provides `address + user + paUrl + database` without a password. The topology check only requires `address && user && !instance`. Consider adding an explicit guard or adjusting `determineTopology` to also require password for PA_PROXY detection. Low risk in practice since CPD signin would fail immediately with wrong credentials.

2. **[src/tests/restService.test.ts] Missing verify propagation test for S2S** - There are verify propagation tests for IAM and CPD, but not for `_authenticateServiceToService`. Consider adding one for completeness since S2S also makes external HTTP calls.

3. **[src/services/RestService.ts:555] `extractCookieValue` regex safety note** - The `name` parameter is interpolated into a RegExp. Currently all callers pass hardcoded strings (`'cam_passport'`, `'ba-sso-csrf'`), so this is safe. If this method were ever exposed or called with user input, it would be a regex injection risk. Consider a brief code comment noting this assumption, or use `name.replace(/[.*+?^${}()|[\\]\\\\]/g, '\\\\$&')` to escape the name defensively.

4. **[src/services/RestService.ts:601] Unreachable status check in `_buildAuthorizationTokenCamSso`** - The `response.status !== 200` check after `axios.get()` is unreachable for 4xx/5xx because axios throws by default on non-2xx. It would only trigger for 201/204 etc. This is a defensive check that mirrors tm1py, so it's fine to keep, but a comment explaining this would be helpful.

5. **[src/services/RestService.ts] Removed `TM1-Namespace` header in Basic auth path** - On main, when `namespace` was set with `user/password` but no `authUrl`, the old code sent `Basic` auth + `TM1-Namespace` header. The PR now routes this to CAM mode with `CAMNamespace` authorization. This appears intentional for tm1py parity. Just flagging as a behavioral change that existing users relying on the old `TM1-Namespace` header behavior should be aware of.

## Strengths

- **Excellent architecture**: The topology-based dispatch (`determineTopology` -> `getAuthenticationMode` -> `setupAuthentication` switch) is much cleaner than the old flat if-else chain. Each auth flow is now a focused, testable method.
- **Strong tm1py parity**: The auth flows (CAMNamespace header, gateway SSO with cam_passport cookie, IAM token exchange, CPD signin, PA Proxy with ba-sso-csrf, S2S with Basic auth + `{User}` POST) all correctly mirror tm1py's implementation.
- **Good helper extraction**: `b64DecodePassword`, `insecureAgentOption`, `normaliseSetCookie`, `extractCookieValue` eliminate duplication while keeping the code readable.
- **Thorough test coverage**: 478 new test lines covering all auth modes, error paths, token exchange flows, TLS verify propagation, and edge cases. The test structure mirrors the code structure well.
- **Proper security practices**: Credentials are Base64-encoded (not logged), `encodeURIComponent` is used for the IAM API key, and `insecureAgentOption` correctly propagates verify settings to all external auth requests.
- **Clean commit history**: Two atomic commits with clear conventional-commit messages referencing the issue number. No AI references.

## Detailed Review

### Verification Steps Completed
- TypeScript compiles cleanly (`tsc --noEmit` -- 0 errors)
- All 119 tests pass (`jest --testPathPattern=restService`)
- No AI references in commit messages
- Removed methods (`setupCamAuthentication`, `setupCamSsoAuthentication`, `setupServiceToServiceAuthentication`) were all private -- no external callers found
- `extractCookieValue` regex parameter is only called with hardcoded strings -- no injection risk
- `encodeURIComponent` properly used for IAM API key in URL-encoded payload
- `insecureAgentOption` correctly propagates `verify` to IAM, CPD, PA Proxy, S2S, and CAM SSO external requests

### Behavioral Changes (Intentional, for tm1py parity)
1. **CAM detection**: No longer requires `authUrl` -- now triggers on `namespace` or `camPassport` alone (matches tm1py)
2. **CAM_SSO**: Now triggers on `gateway` (was: `authUrl + user + password + namespace`)
3. **CAM auth header**: Uses `CAMNamespace user:pass:ns` Base64 header instead of JSON POST (matches tm1py)
4. **Gateway SSO**: GET to gateway with `CAMNamespace` param, extracts `cam_passport` cookie (new, matches tm1py)
5. **S2S flow**: Uses Basic auth with `clientId:clientSecret` + POST `{User}` instead of `client_credentials` grant (matches tm1py)
6. **IBM Cloud**: Topology-based detection via `iamUrl` instead of `apiKey + user==='apikey'`
7. **`TM1-Namespace` header removed**: Namespace now routes to CAM, not Basic+header

## Review Sources
- Manual Review (Stream A): 1 P1 issue, 5 P2 suggestions
- Note: Multi-perspective agent review (Stream B) was not available (Task tool unavailable in this session)

## Next Steps
- [ ] Fix P1: Add namespace validation for CAM_SSO (either in detection or in setupAuthentication)
- [ ] Consider P2 suggestions
- [ ] Re-request review after fix

File: src/services/RestService.ts [hunk 2]
```diff
@@ -513,192 +513,348 @@ export class RestService {
...
+
+            case AuthenticationMode.SERVICE_TO_SERVICE:
+                await this._authenticateServiceToService();
+                break;
+
+            case AuthenticationMode.CAM_SSO: {
+                // CAM_SSO is only reached when gateway is set (see getAuthenticationMode)
+                const token = await RestService._buildAuthorizationTokenCamSso(
+                    this.config.gateway!,
+                    this.config.namespace!,
```
Comment (hsubebe89) [inline src/services/RestService.ts:824]: **P1 (related): Non-null assertion on `namespace` is unsafe**

`this.config.namespace!` suppresses TypeScript's null check, but `namespace` is not guaranteed to be set when `CAM_SSO` is selected (only `gateway` is checked in `getAuthenticationMode()`). If the detection is fixed to require namespace, this assertion becomes safe. Otherwise, add an explicit guard here.

File: src/services/RestService.ts [hunk 3]
```diff
@@ -513,192 +513,348 @@ export class RestService {
...
-            return;
+    /**
+     * Extract a named cookie value from raw Set-Cookie headers.
+     */
+    private static extractCookieValue(
+        raw: string | string[] | undefined,
+        name: string
+    ): string | undefined {
+        for (const header of RestService.normaliseSetCookie(raw)) {
+            const match = header.match(new RegExp(`${name}=([^;]+)`));
```
Comment (hsubebe89) [inline src/services/RestService.ts]: **P2: Regex injection note**

The `name` parameter is interpolated into `new RegExp(...)`. Currently all callers pass hardcoded strings (`'cam_passport'`, `'ba-sso-csrf'`), so this is safe. Consider adding a brief comment noting this assumption, or defensively escaping the name:
```typescript
const escaped = name.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
const match = header.match(new RegExp(`${escaped}=([^;]+)`));
```

File: src/services/RestService.ts [hunk 4]
```diff
@@ -513,192 +513,348 @@ export class RestService {
...
+            case AuthenticationMode.IBM_CLOUD_API_KEY: {
+                const accessToken = await this._generateIbmIamCloudAccessToken();
+                this.axiosInstance.defaults.headers.common['Authorization'] =
+                    `Bearer ${accessToken}`;
+                break;
+            }
+
+            case AuthenticationMode.PA_PROXY: {
+                const jwt = await this._generateCpdAccessToken({
+                    username: this.config.user!,
```
Comment (hsubebe89) [inline src/services/RestService.ts]: **P2: `password!` non-null assertion may be unsafe**

The PA_PROXY topology is detected by `address && user && !instance` (in `determineTopology`), which does not require `password`. If a user provides the PA_PROXY config without a password, `password!` would be `undefined`, causing a confusing error at the CPD signin endpoint.

Consider adding a guard:
```typescript
case AuthenticationMode.PA_PROXY: {
    if (!this.config.user || !password) {
        throw new Error('PA Proxy authentication requires user and password');
    }
    // ...
}
```

File: src/services/RestService.ts [hunk 5]
```diff
@@ -513,192 +513,354 @@ export class RestService {
...
+    private static async _buildAuthorizationTokenCamSso(
+        gateway: string,
+        namespace: string,
+        verify?: boolean | string
+    ): Promise<string> {
+        const response = await axios.get(gateway, {
+            params: { CAMNamespace: namespace },
+            ...RestService.insecureAgentOption(verify)
+        });
+        if (response.status !== 200) {
```
Comment (hsubebe89) [inline src/services/RestService.ts:604]: **[P2 - Suggestion]** This `response.status !== 200` check is unreachable for 4xx/5xx responses because axios throws an `AxiosError` for non-2xx status codes by default. It only catches non-200 success codes (201, 204, etc.).

If you want the custom error message to fire for all non-200 responses, add `validateStatus: () => true` to the axios config:
```ts
const response = await axios.get(gateway, {
    params: { CAMNamespace: namespace },
    validateStatus: () => true,
    ...RestService.insecureAgentOption(verify)
});
```

Alternatively, a comment noting that axios throws before this check would suffice. Not blocking -- current behavior is safe since the axios error propagates.

File: src/services/RestService.ts [hunk 6]
```diff
@@ -513,192 +513,354 @@ export class RestService {
...
-            throw new Error(`CAM SSO authentication failed: ${error}`);
+    /**
+     * Authenticate with PA Proxy using a CPD JWT token.
+     * Mirrors tm1py's PA_PROXY flow in _start_session.
+     */
+    private async _authenticateWithPaProxy(jwt: string): Promise<void> {
+        const authRoot = this.resolveRoots().authRoot;
+        const headers = { 'Content-Type': 'application/x-www-form-urlencoded' };
+        const payload = `jwt=${jwt}`;
+        const response = await axios.post(authRoot, payload, {
```
Comment (hsubebe89) [inline src/services/RestService.ts:676]: **[P2 - Suggestion]** The JWT is interpolated directly into the URL-encoded payload without escaping. While JWTs are base64url-encoded (no `+`, `=`, `&` characters), `encodeURIComponent(jwt)` would be marginally more defensive:
```ts
const payload = `jwt=${encodeURIComponent(jwt)}`;
```
This matches how `apiKey` is encoded in `_generateIbmIamCloudAccessToken` (line 630). tm1py also doesn't encode the JWT, so this is parity -- just a consistency suggestion.

**Review by hsubebe89 — COMMENTED** [review]
## Summary

This PR rewrites the authentication layer in RestService.ts to align with tm1py's auth flows. It replaces ad-hoc if/else auth dispatch with topology-aware mode detection and a clean switch-based `setupAuthentication()`, adds new auth flows (IBM Cloud IAM, CPD, PA Proxy, Gateway SSO), and fixes S2S to use Basic auth with `{User: user}` POST (matching tm1py). The refactoring extracts 5 shared static helpers and adds 40+ well-structured tests covering all auth paths.

## Status: APPROVED (posted as COMMENT because GitHub blocks self-approval)

## Scores
- Security: 8/10
- Performance: 9/10
- Code Quality: 9/10
- Test Coverage: 9/10

## Review Sources
- Manual Review (Stream A): 0 P0, 0 P1, 5 P2
- Multi-Perspective Review (Stream B, 5-agent): 0 issues above confidence threshold

---

## Parity Verification (tm1py reference check)

I verified each auth flow against [tm1py/Services/RestService.py](https://github.com/cubewise-code/tm1py/blob/master/TM1py/Services/RestService.py) on master:

| Auth Mode | tm1py behavior | tm1npm PR behavior | Match? |
|-----------|---------------|-------------------|--------|
| BASIC | `_build_authorization_token_basic` -> `Basic base64(user:password)` | `_buildAuthorizationTokenBasic` -> same | Yes |
| CAM (passport) | `CAMPassport {passport}` header | Same | Yes |
| CAM (namespace) | `_build_authorization_token_cam` -> `CAMNamespace base64(user:password:namespace)` | `_buildAuthorizationTokenCam` -> same | Yes |
| CAM_SSO (gateway) | GET gateway with `CAMNamespace` param, extract `cam_passport` cookie, return `CAMPassport` header | Same (note: omits HttpNegotiateAuth/NTLM -- documented in JSDoc) | Yes (with documented limitation) |
| IBM_CLOUD_API_KEY | POST to `iam_url` with URL-encoded grant_type+apikey, extract `access_token`, set Bearer | Same (additionally URL-encodes apiKey -- improvement) | Yes |
| PA_PROXY | Generate CPD token via POST to `{cpd_url}/v1/preauth/signin`, then POST `jwt={jwt}` to auth root, extract `ba-sso-csrf` cookie, set `ba-sso-authenticity` header | Same | Yes |
| SERVICE_TO_SERVICE | `HTTPBasicAuth(client_id, secret)` + POST `{"User": user}` to auth endpoint, extract session cookie (with domain fallback) | Same (Basic auth header + POST JSON, parseSetCookieHeaders handles domain-agnostic extraction) | Yes |
| ACCESS_TOKEN | `Bearer {access_token}` header | Same | Yes |
| BASIC_API_KEY | tm1py: `user == 'apikey' and saas.ibm.com in base_url` | tm1npm: `apiKey` config field exists | Intentional divergence -- tm1npm has dedicated apiKey config field |
| WIA | HttpNegotiateAuth (Windows-only) | Throws with clear error message | Correct -- Node.js doesn't have SSPI |
| `_determine_auth_mode` | Topology-first for v12, then config-flag fallback for v11 | `getAuthenticationMode()` uses same topology-first pattern | Yes |
| `b64_decode_password` | `b64decode(pw).decode("UTF-8")` | `Buffer.from(pw, 'base64').toString('utf-8')` | Yes |
| `TM1-Namespace` header (old) | Not present in tm1py | Correctly removed by this PR | Yes |

**Verdict: All auth flows match tm1py behavior. The PR achieves its stated goal of parity.**

---

## Issues Found

### P0 - Critical
None.

### P1 - Important
None.

### P2 - Suggestions (Optional)

1. **[src/services/RestService.ts:604]** `_buildAuthorizationTokenCamSso` status check is unreachable for error codes [Manual]
   - The `if (response.status !== 200)` check will never trigger for 4xx/5xx responses because axios throws an `AxiosError` for non-2xx by default. The check only catches non-200 success codes (201, 204, etc.). Consider adding `validateStatus: () => true` to the axios config if you want the custom error message to surface, or note in a comment that axios throws before reaching this check. Not blocking -- current behavior is safe.

2. **[src/services/RestService.ts:600]** Gateway SSO: axios follows redirects by default [Manual]
   - `axios.get(gateway, ...)` will follow HTTP redirects by default. tm1py's `requests.get` also follows redirects by default, so this is parity. Just noting for awareness -- if a gateway issues a redirect chain, the `set-cookie` from intermediate responses may be lost. Consider `maxRedirects: 0` if the gateway is expected to return the passport cookie on the initial response. Non-blocking.

3. **[src/services/RestService.ts:676]** `_authenticateWithPaProxy` JWT not URL-encoded [Manual]
   - The JWT is interpolated directly into `jwt=${jwt}`. While JWTs are base64url-encoded and safe in practice, `encodeURIComponent(jwt)` would be more defensive and consistent with how `apiKey` is encoded in `_generateIbmIamCloudAccessToken` (line 630). tm1py also doesn't encode, so this is parity. Non-blocking.

4. **[src/services/RestService.ts:531-536]** `insecureAgentOption` creates a new `https.Agent` per call [Manual]
   - Each auth request that passes `verify: false` creates a fresh `https.Agent`. Since auth only runs once (or on re-auth), this is negligible. For future optimization, consider caching. Non-blocking.

5. **[src/tests/restService.test.ts:1246-1295]** PA_PROXY tests don't verify `paSession` cookie is captured [Manual]
   - The PA_PROXY test verifies `ba-sso-authenticity` header and CPD+proxy POST calls, but doesn't assert that the `paSession` cookie was captured in `sessionCookies`. The S2S test at line 1336 does verify cookie capture. Consider adding a similar assertion. Non-blocking.

---

## Strengths

- **Excellent parity with tm1py**: Every auth flow was carefully ported with correct token formats, HTTP methods, headers, and cookie handling. The topology-based auth mode detection mirrors tm1py's `_determine_auth_mode` faithfully.
- **Clean refactoring**: Extracting `insecureAgentOption`, `normaliseSetCookie`, `extractCookieValue`, `_buildAuthorizationTokenBasic`, and `_buildAuthorizationTokenCam` as static helpers eliminates duplication and makes each auth flow readable.
- **Well-documented limitations**: The JSDoc on `_buildAuthorizationTokenCamSso` clearly explains that NTLM/Kerberos is not supported in Node.js and suggests using `camPassport` as a workaround.
- **Defensive validation**: Guards in `PA_PROXY` (user/password required), `CAM_SSO` (namespace required alongside gateway), and `S2S` (authUrl required for v11 topology) prevent silent misconfiguration.
- **Thorough test coverage**: 40+ new tests cover all auth modes, error paths, B64 password decoding, verify propagation, cookie capture, and edge cases like gateway-without-namespace fallback.
- **Clean commit history**: Three atomic commits (implement -> refactor -> harden) with clear conventional commit messages and no AI references.

## Detailed Analysis

### Architecture
The switch from if/else chain to topology-first `getAuthenticationMode()` + switch-based `setupAuthentication()` is a significant improvement. The old code had implicit priority ordering that was hard to reason about. The new code makes the dispatch explicit and matches tm1py's routing.

### Security
- `verify` is correctly propagated to all external auth requests via `insecureAgentOption`.
- No credentials are logged or exposed in error messages.
- Cookie extraction uses `indexOf`/`split` rather than regex (hardened in commit 3).
- `parseSetCookieHeaders` sanitizes CR/LF/NUL characters (pre-existing defense, preserved by this PR).

### Breaking Changes (All Correctness Fixes)
- Old `setupCamAuthentication` used JSON POST with `{parameters: [{name: 'CAMPassport', value: ...}]}` -- not tm1py-compatible. Now correctly uses `Authorization: CAMPassport` header.
- Old `setupCamSsoAuthentication` used JSON POST with `{username, password, namespace}` -- not tm1py-compatible. Now correctly uses `Authorization: CAMNamespace base64(...)` header.
- Old `setupServiceToServiceAuthentication` used `grant_type: client_credentials` OAuth2 flow -- does not match tm1py. Now correctly uses Basic auth + `{User: user}` POST.
- These are all correctness fixes, not regressions.

### Build & Tests
- TypeScript compiles cleanly (`tsc --noEmit` exit 0)
- All 121 tests pass (including 40+ new auth flow tests)

## Next Steps
- [ ] Consider P2 suggestions (all optional)
- [ ] Merge when CI passes
